sdk/src/types/block/core/wrapper.rs

```diff
@@ -0,0 +1,348 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::mem::size_of;
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use packable::{
+    error::{UnexpectedEOF, UnpackError, UnpackErrorExt},
+    packer::{Packer, SlicePacker},
+    unpacker::{CounterUnpacker, SliceUnpacker, Unpacker},
+    Packable, PackableExt,
+};
+
+use crate::types::block::{
+    block_id::{BlockHash, BlockId},
+    core::{BasicBlock, ValidationBlock},
+    protocol::ProtocolParameters,
+    signature::Signature,
+    slot::{SlotCommitmentId, SlotIndex},
+    Block, Error, IssuerId,
+};
+
+/// Represent the object that nodes gossip around the network.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BlockWrapper {
+    /// Protocol parameters of the network to which this block belongs.
+    // TODO Not fully sure this is worth the small UX improvement, needs to be discussed more.
```

Comment:
issue? Although last time we discussed it was a yes

Reply:
https://github.com/iotaledger/iota-sdk/issues/1224